gulpfile.js

```diff
@@ -82,145 +83,124 @@ const dev_plugins = Array.from(
+  const version = get_version(pluginName);
+  if (!pluginName) {
+    fs.writeFileSync(`${src}/src/version.js`, `/* Generated by gulpfile.js, do not edit manually */\n\nexport default '${version}';`);
+  }
   fs.readFile(`${src}/README.md`, 'utf8', function (_, data) {
     data = (data || '').toString().split("\n");
     data.splice(0, 1, pluginName ? `# g3w-client-plugin-${pluginName} v${version}` : `# G3W-CLIENT v${version}`);
     fs.writeFile(`${src}/README.md`, data.join("\n"), 'utf8', (err) => { if (err) return console.log(err); });
   });
 };
 
+setNODE_ENV();
+
 /**
- * Compile plugin.js and app.js 
- * 
  * @param { string } pluginName name of plugin to build (eg. 'editing')
  * @param { boolean } watch     whether to watchify source files
- * @param ctx                   task context
- *
+ * 
  * @since 3.10.0
  */
-async function browserify_plugin(pluginName, watch = true, ctx = false) {
-  const is_client = 'client' === pluginName;
-
-  let src          = `${g3w.pluginsFolder}/${pluginName}`;                // plugin folder (git source)
-  let outputFolder = production
+const browserify_plugin = (pluginName, watch = true) => {
+  const src          = `${g3w.pluginsFolder}/${pluginName}`;              // plugin folder (git source)
+  const outputFolder = production
     ? `${g3w.admin_plugins_folder}/${pluginName}/static/${pluginName}/js/`// plugin folder (PROD env)
     : `${g3w.admin_overrides_folder}/static/${pluginName}/js/`;           // plugin folder (DEV env)
 
   console.log(INFO__ + `Building plugin:` + __RESET + ' → ' + outputFolder);
 
-  const path = is_client ? '.' : `${g3w.pluginsFolder}/${pluginName}`;
-  const _src = is_client ? `./src/index.${production ? 'prod' : 'dev'}.js` : `./${pluginName}/index.js`;
-
-  const opts = {
-    basedir: is_client ? './'                                            : `${g3w.pluginsFolder}`,
-    paths:   is_client ? ['./src/', './src/app/']                        : [`${g3w.pluginsFolder}`],
+  let bundler = browserify(`./${pluginName}/index.js`, {
+    basedir: `${g3w.pluginsFolder}`,
+    paths: [`${g3w.pluginsFolder}`],
     debug: !production,
     cache: {},
     packageCache: {},
-    insertGlobals: true,
-    insertGlobalVars: {
-      // inject plugin version (commit + branch). In case of new release (tag) no branch is added to version
-      __git__: () => {
-        try {
-          let branch = execSync(`git -C  ${path} rev-parse --abbrev-ref HEAD`, { encoding: 'utf8' }).trim();
-          if (branch && 'HEAD' !== branch.trim()) {
-            return JSON.stringify({ branch, commit: execSync(`git -C  ${path} log -n 1 --pretty=format:"%H"`, { encoding: 'utf8' }).trim().substring(0,7) });
-          }
-        } catch(err) {
-          console.warn(YELLOW__ + '[WARN] ' + __RESET + 'git info not found (' + GREEN__ + path + __RESET + ')' );
-        }
-        return null;
-      },
-    },
```

Comment:
@volterra79 I removed the `insertGlobalVars` solution because:

1. it generates a lot of duplicate and unintelligible code
2. it is significantly slower than `gulp-replace`
3. the overall build process was no longer synchronous (ref: `build:client` task + `runSequence` and any nested `await`/`async` function)